diff --git a/configure.ac b/configure.ac
index 9792cba..359cc29 100644
--- a/configure.ac
+++ b/configure.ac
@@ -144,11 +144,13 @@ esac
 
 # Set some variables for building build platform helpers.
 case "$build_os" in
-    *zOS*)
-        EXTRA_LIBS_FOR_BUILD=zoslib
+    *openedition*)
+        PKG_CHECK_MODULES([ZOSLIB], [zoslib],
+                          [EXTRA_LIBS_FOR_BUILD="$ZOSLIB_LIBS"],
+                          [AC_MSG_ERROR([zoslib not found. Please ensure zoslib.pc is in PKG_CONFIG_PATH.])])
         ;;
     *)
-        EXTRA_LIBS_FOR_BUILD=
+        EXTRA_LIBS_FOR_BUILD=""
 	;;
 esac
 AC_SUBST(EXTRA_LIBS_FOR_BUILD)
diff --git a/src/spawn-posix.c b/src/spawn-posix.c
index 8cdd032..73cef10 100644
--- a/src/spawn-posix.c
+++ b/src/spawn-posix.c
@@ -63,7 +63,7 @@
 struct gpgrt_spawn_actions {
   int fd[3];
   const int *except_fds;
-  char **environ;
+  char **env;
   const char *const *envchange;
   void (*atfork) (void *);
   void *atfork_arg;
@@ -432,8 +432,8 @@ my_exec (const char *pgmname, const char *argv[], gpgrt_spawn_actions_t act,
   if (pgmname == NULL)
     return 0;
 
-  if (act->environ)
-    execve (pgmname, (char *const *)argv, act->environ);
+  if (act->env)
+    execve (pgmname, (char *const *)argv, act->env);
   else
     execv (pgmname, (char *const *)argv);
 
@@ -534,7 +534,7 @@ void
 _gpgrt_spawn_actions_set_environ (gpgrt_spawn_actions_t act,
                                   char **environ_for_child)
 {
-  act->environ = environ_for_child;
+  act->env = environ_for_child;
 }
 
 void
